pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -242,3 +262,133 @@ func (b *BridgePodInterface) createDefaultBridge() error {
 
 	return nil
 }
+
+type SlirpPodInterface struct {
+	iface       *v1.Interface
+	network     *v1.Network
+	domain      *api.Domain
+	slirpConfig api.Arg
+}
+
+func (p *SlirpPodInterface) setup() error {
+	commandLine, err := getCachedQemuCommandLine()
+	if err != nil {
+		return err
+	}
+
+	if commandLine != nil {
+		p.domain.Spec.QEMUCmd = commandLine
+
+		return nil
+	}
+
+	err = p.configVMCIDR()
+	if err != nil {
+		return err
+	}
+
+	err = p.configDNSSearchName()
+	if err != nil {
+		return err
+	}
+
+	err = p.configPortForward()
+	if err != nil {
+		return err
+	}
+
+	err = p.commitConfiguration()
+	if err != nil {
+		return err
+	}
+
+	err = setCachedCommandLine(p.domain.Spec.QEMUCmd)
+	if err != nil {
+		panic(err)
+	}
+	return nil
+}
+
+func (p *SlirpPodInterface) configPortForward() error {
+	if p.iface.Slirp.Ports == nil {
+		return nil
+	}
+
+	portForwardMap := make(map[int32]string)
+
+	for _, forwardPort := range p.iface.Slirp.Ports {
+		protocol := DefaultProtocol
+
+		if forwardPort.Port == 0 {
+			return fmt.Errorf("Port must be configure")
+		}
+
+		// Check protocol, its case sensitive like kubernetes
```

Comment:
This validation code is more or less the same as in webhook package. I suggest we move it into a function and reuse from both places.

Reply:
I will just remove it from the virt-lancher, Because we already check the configuration in the virt-api and I dont see a reason to recheck it inside the virt lancher

Reply:
Defensive programming; the approach in general would also help with upgrades when different components move separately and may incorporate different rules. But yes I have no problem with removing it for now, we'll need a more thought through approach to all that not specific to this PR.